components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -4057,56 +4059,60 @@ public void deleteOAuthApplication(String consumerKey) throws APIManagementExcep
+                        consumerKey = ((APIKey) ((ArrayList) application.getKeys()).get(i)).getConsumerKey();
+                        OAuthApplicationInfo oAuthApplicationInfo = keyManager.retrieveApplication(consumerKey);
+                        OAuthAppRequest oauthAppRequest = ApplicationUtils.createOauthAppRequest(oAuthApplicationInfo.
+                                        getParameter(ApplicationConstants.OAUTH_CLIENT_NAME).toString(), null,
+                                oAuthApplicationInfo.getCallBackURL(), null,
+                                null, application.getTokenType());
+                        oauthAppRequest.getOAuthApplicationInfo().setAppOwner(userId);
+                        oauthAppRequest.getOAuthApplicationInfo().setClientId(consumerKey);
+                             /* updating the owner of the OAuth application with userId */
+                        OAuthApplicationInfo updatedAppInfo = keyManager.updateApplicationOwner(oauthAppRequest,
+                                oldUserName);
+                    }
+                    isAppUpdated = true;
+                    audit.info("Successfully updated the owner of application " + application.getName() +
+                            " from " + oldUserName + " to " + userId);
+                } else {
+                    throw new APIManagementException("Unable to update application owner to " + userId +
+                            " as this user has a application with the same name. Update owner to another user");
                 }
             }
-
-            String[] roleArr = roleList.toArray(new String[roleList.size()]);
-
-            APIManagerConfiguration config = getAPIManagerConfiguration();
-            String serverURL = config.getFirstProperty(APIConstants.AUTH_MANAGER_URL) + "UserAdmin";
-            String adminUsername = config.getFirstProperty(APIConstants.AUTH_MANAGER_USERNAME);
-            String adminPassword = config.getFirstProperty(APIConstants.AUTH_MANAGER_PASSWORD);
-
-            UserAdminStub userAdminStub = new UserAdminStub(serverURL);
-            CarbonUtils.setBasicAccessSecurityHeaders(adminUsername, adminPassword, userAdminStub._getServiceClient());
-            userAdminStub.updateRolesOfUser(userId, roleArr);
-            isAppUpdated = true;
-
-        } catch (org.wso2.carbon.user.api.UserStoreException e) {
-            handleException("Error when getting the tenant's UserStoreManager or when getting roles of user ", e);
-        } catch (RemoteException e) {
-            handleException("Server couldn't establish connection with auth manager ", e);
-        } catch (UserAdminUserAdminException e) {
-            handleException("Error when getting the tenant's UserStoreManager or when getting roles of user ", e);
+        } else {
+            throw new APIManagementException("Unable to update application owner to " +
+                    userId + " as this user does not belong to this tenant domain");
```

Comment:
shouldn't we add the tenant domain information to the log message?

Reply:
Tenant domain information is added